src/main/java/teammates/logic/backdoor/BackDoorLogic.java

```diff
@@ -555,8 +548,11 @@ public void removeDataBundle(DataBundle dataBundle) {
         // We don't attempt to delete them again, to save time.
         deleteCourses(dataBundle.courses.values());
 
-        populateNullStudentProfiles(dataBundle.accounts.values());
         accountsDb.deleteAccounts(dataBundle.accounts.values());
+        // delete associated profiles
+        // TODO: Remove the following line after tests have been run against staging server
```

Comment:
What exactly needs to be done here?

Reply:
The new behaviour introduced here is to create profile only when needed. The previous implementation create empty profile for every account.

So `instructorStudentRecordsPageMixedQuestionType.html` and `instructorStudentRecordsPageWithPrivateFeedback.html` is modified because the corresponding profile for the account is not created (not in data bundle). Previously an empty profile is created for every account and that is why is it displayed.

Therefore, without this line, the test against LIVE server in pre-deployment check will fail.